src/goto-instrument/contracts/contracts.cpp

```diff
@@ -1016,6 +1018,13 @@ void code_contractst::apply_loop_contract(
     goto_functions.update();
     natural_loops_mutablet updated_loops(goto_function.body);
 
+    // We will unwind all transformed loops twice. Store the names of
+    // to-unwind-loops here and perform the unwinding after transformation done.
```

Comment:
It's missing the "why". Let's document here why we need to unroll the loops twice.

Reply:
New comments added to explain why we need to unroll transformed loops.